snapshots/windows/windows.go

```diff
@@ -202,14 +205,58 @@ func (s *snapshotter) Commit(ctx context.Context, name, key string, opts ...snap
 	// grab the existing id
 	id, _, _, err := storage.GetInfo(ctx, key)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "failed to get storage info for %s", key)
 	}
 
-	usage, err := fs.DiskUsage(ctx, s.getSnapshotDir(id))
+	snapshot, err := storage.GetSnapshot(ctx, key)
 	if err != nil {
 		return err
 	}
 
+	path := s.getSnapshotDir(id)
+
+	if strings.Contains(key, snapshots.UnpackKeyPrefix) {
```

Comment:
This path used to contain code, before changes in `ociwclayer.ImportLayerFromTar`, and before `UnpackKeyPrefix` was introduced to allow a one-line identification of imported layers.

Also, see https://github.com/containerd/containerd/pull/4419/commits/f1f6432b693e6c8baa63ffd5c1ac9a839fc048e2#diff-b6ae335f862c747466f1bc00e89eb1ef7cd9589bcc89ec57953c9c20125b3752R218-R224 where I need to insert another case in here, and this lets me avoid nesting most of the rest of the function an extra level of indent, or having a chain like `if !strings.Contains(key, snapshots.UnpackKeyPrefix) && condition == 1 {} else if !strings.Contains(key, snapshots.UnpackKeyPrefix) && condition == 2`.

I realise it's a bit weird, but I personally like explicit "In this specific case, ignore the following code blocks" when the "following code-block" is itself somewhat complex. I don't like it enough to refuse to change it if reviewers find it unclear, though. ^_^

I think of it like an early return in a function, which is quite reasonable, except we're not leaving the function yet.

Another option (which I actually like, now I think about it) is moving the 'reimport' logic into a separate function. I have a vague idea that it might be worth implementing this reimport logic in hcsshim and just calling it, in case future improvements to the WCOW filesystem structure allow this to be more efficient, so making it a distinct function here will be illustratively useful.